lib/validation/validators.js

```diff
@@ -397,7 +399,9 @@ function validateReferences (api) {
   // Identify references and validate missing references for non-JSON References (security)
   _.forEach(api.definitionFullyResolved.security, createSecurityProcessor(['security']));
 
-  _.forEach(api.definitionFullyResolved.paths, function (pathDef, name) {
+  var pathObjects = stripExtensions(api.definitionFullyResolved.paths);
```

Comment:
Wouldn't it make sense to just test `pathDef` for "x-" within the `_.forEach` loop (here and the other places they are used), instead of iterating over items multiple times with `stripExtensions` and `_.forEach`?